packages/core/components/NestedDropdown/NestedDropdown.tsx

```diff
@@ -0,0 +1,262 @@
+import { useState, useEffect, useRef, useMemo } from 'react'
+import './nesteddropdown.styles.css'
+import Icon from '@cdc/core/components/ui/Icon'
+
+const Options: React.FC<{
+  currentOptions: (string | number)[]
+  label: string
+  handleSecondTierSelect: Function
+  userSelectedTierTwoLabel: string
+  userSearchTerm: string
+}> = ({ currentOptions, label, handleSecondTierSelect, userSelectedTierTwoLabel, userSearchTerm }) => {
+  const [isTierOneExpanded, setIsTierOneExpanded] = useState(true)
+
+  const checkMark = <>&#10004;</>
+
+  useEffect(() => {
+    setIsTierOneExpanded(userSearchTerm.length > 0 ? true : isTierOneExpanded)
+  }, [userSearchTerm])
+
+  const handleGroupClick = e => {
+    const leaveExpanded = e.target.className === 'selectable-item' ? true : !isTierOneExpanded
+    setIsTierOneExpanded(leaveExpanded)
+  }
+
+  const handleSecondTierClick = (tierTwo: string | number) => {
+    handleSecondTierSelect(tierTwo)
+  }
+
+  const handleKeyUp = (e: { target: any; key: string }) => {
+    const currentItem = e.target
+    if (e.key === 'ArrowRight') setIsTierOneExpanded(true)
+    else if (e.key === 'ArrowLeft') {
+      if (currentItem.className === 'selectable-item') currentItem.parentNode.parentNode.focus()
+      setIsTierOneExpanded(false)
+    } else if (e.key === 'Enter') {
+      currentItem.className === 'selectable-item' ? handleSecondTierClick(currentItem.dataset.value) : setIsTierOneExpanded(!isTierOneExpanded)
+    }
+  }
+
+  return (
+    <>
+      <li role='treeitem' key={label} tabIndex={0} aria-label={label} onClick={handleGroupClick} onKeyUp={handleKeyUp} className={`nested-dropdown-group${isTierOneExpanded ? '' : '-collapsed'}`}>
```

Comment:
why not just use the .hide class that you already created?

Reply:
I am using hide now. I still needed to use the "-collapse" for the Arrow logic but it has no effect on the display anymore.

Reply:
Don’t use class to describe state only try checking for the “hide” property instead or use data attribute

Reply:
The hide property is on it's child. I didn't want to through another .firstChild into the mix but I can.

Reply:
I think that's an acceptable option. it's more clear from the js side what you're looking for.

Reply:
Just for clarity, I should change the code to not depend on nested-dropdown-group-collapse"